docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+    > "add" is detected.
+
+3.  addAppointment() method in the AppointmentActions.java class is
+    > called and adds the details into the list
+
+[Classes:]{.ul}
+
+1.  AppointmentActions
+
+2.  DrugActions
+
+3.  NurseScheduleActions
+
+4.  PatientList
+
+5.  StaffList
+
+### **Object Class component**
+
+This component is to distinguish the separate object that has been
+instantiated to hold the details of our target users. An object class
+allows for a structure that helps store details from user input in the
+format that we want to be saved.
+
+[Objects:]{.ul}
+
+1.  DoctorAppointment.java
+
+2.  Drug.java
+
+3.  Patient.java
+
+4.  NurseSchedule.java
+
+5.  Staff.java
+
+### 
+
+### **Storage component**
+
+The difference classes for storages are separated into their respective
+architecture for easier management.
+
+> [Methods:]{.ul}
+
+1.  writeToFile() - Calling upon this method whenever there is change to
+    > the data array list.
+
+2.  createFile() - Called during the start of the program when
+    > loadFile() is called and returns an error, meaning that there are
+    > no text files in storage. Thus, this method is called to create a
+    > new text file for data storage.
+
+3.  loadFile() - Called during the start of the program, to initialise
+    > the array list and load previously saved data.
+
+[Classes:]{.ul}
+
+1.  DoctorAppointmentStorage
+
+2.  DrugStorage
+
+3.  NurseScheduleStorage
+
+4.  PatientStorage
+
+5.  StaffStorage
+
+### **Common classes**
+
+1.  Main UI Class
+
+2.  Main Storage Class
+
+ 
+
+**Implementation**
+------------------
+
+This section describes how the features are being implemented. The scope
+will include the Doctor Appointments / Drugs / Nurse Schedules /
+Patients / Staff related features.
+
+Common features that can be identified across the scope include these
+distinct functionalities: Adding, Deleting and Listing. For Nurse
+Schedules/ Patients / Staff related features, there is an additional
+search functionality.
+
+### **Menu-related Features**
+
+**Main Menu**
+
+This feature allows users to select the different menus for Patient /
+Staff / Doctor Appointments / Nurse Schedules and Drugs through
+inputting commands within the main menu. Failure to input a correct or
+recognized command will then prompt the program to generate an exception
+that will alert the user of what they have done wrongly, and also prompt
+the user to access the help list to view the correct set of commands and
+their proper syntaxes.
+
+If the user enters a correct and recognized command, they will be able
+to access the functionalities related to the specific Objects (i.e
+Staff, Patient, Doctor Appointments, Nurse Schedules, Drugs).
+
+[Implementation:]{.ul}
+
+1.  A command is requested from the user as the application starts,
+    > Duke.run() calls UI.scanInput() to read the user's input from the
+    > command line.
+
+2.  The user input determines which type of functionalities will be
```

Comment:
Good, comprehensive and detailed explanation for the general running of the application :)